src/runtime/recipe/handle-connection.ts

```diff
@@ -169,43 +169,47 @@ export class HandleConnection {
   isResolved(options?): boolean {
     assert(Object.isFrozen(this));
 
-    if (this.isOptional) {
-      return true;
-    }
-
     let parent;
     if (this.spec && this.spec.parentConnection) {
       parent = this.particle.connections[this.spec.parentConnection.name];
+      if (!parent || !parent.handle) {
+        if (options) {
+          options.details = 'parent connection missing handle';
+        }
+        return false;
+      }
     }
 
-    // TODO: This should use this._type, or possibly not consider type at all.
-    if (!this.type) {
+    if (parent && !parent.handle) {
+      // Our parent doesn't resolve, we don't need to resolve.
```

Comment:
* if our parent is optional, it could still resolve, couldn't it?
* should we expect to have a parent at the recipe level if the parent doesn't connect to a handle? I thought we avoided instantiating connections unless they were connected?

Reply:
To summarize our conversation, I believe that we decided that the statement of a handle-connection inside a particle is stating that the connection should be connected.